### ISSUE TITLE
BackwardsCompatiblity wrap_arity not allowing (-1) arity.

### ISSUE THREAD

**Issue by abMatGit** [rmosolgo/graphql-ruby#979]
I'm not entirely sure if this is an anti-pattern of implementing `resolve` or `resolve_type` however we have code resembling:
```ruby
Schema = GraphQL::Schema.define do
  ...
  resolve_type ->(*) { nil }
end
```
This is hitting the [else condition of your `wrap_arity` case check](https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/backwards_compatibility.rb#L28-L30).

Proposal: 
```ruby
case arity
when -1
  # It already matches, return it as is
  callable
when to
...
end
```

**Comment by rmosolgo** [comment]
:+1: sounds like a good solution!

**Comment by rmosolgo** [comment]
This was addressed in #1037 : 

https://github.com/rmosolgo/graphql-ruby/pull/1037/files#diff-2b899a3d0d7c3327183e41ec5882408fR15

So it will be fixed in 1.8.0 (currently available as a prerelease)!
